### PR TITLE
fix train epoch metrics

### LINKED ISSUE #3228: Log epoch as step when on_epoch=True and on_step=False
## 🚀 Feature

When using the new structured `Result` API, it is no longer possible to force PL to report the epoch as the current step to loggers instead of the global step (or elapsed number of training steps).

### Motivation

 This results in confusing results when viewing results in a tool that uses step count by default (e.g. tensorboard's scalars view). Intuitively, one would think `.log(..., on_epoch=True, on_step=False)` would count epochs and not steps.

It is possible to obtain this behaviour by overriding both `training_epoch_end` _and_ `validation_epoch_end` and returning an `EvalResult` with a "step" metric. Unfortunately, this adds back a bunch of boilerplate and many of the nice metric aggregation features PL offers when `*_epoch_end` is not implemented.

### Pitch

Either a) allow for overriding the step for a given result, or b) default that step to `current_epoch` when `on_epoch=True` and `on_step=False`.

### Alternatives

1. Use a "step" key with the old dict-based logging system (not documented, but worked as of 0.8.3)
2. Override train/validation_epoch_end and add return an `EvalResult` with `.log('step', self.current_epoch)`.

### Additional context

Original discussion: https://forums.pytorchlightning.ai/t/is-there-a-way-to-only-log-on-epoch-end-using-the-new-result-apis/74/3

### PR DESCRIPTION
**Pull request by beyondyoni**
## What does this PR do?

fix #3228 

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by beyondyoni** [comment]
@williamFalcon Thanks for assisting!
as you see, I only made 2 very small changes. For some reason I'm trying to understand, they cause the CI tests to crash

**Comment by williamFalcon** [comment]
yes lol... this is what i'm saying... it's not as simple as just 2 changes.

This is a much deeper thing that touches a lot. I'm afraid this has to wait until after 1.0.

We can't add the instability to the package.
